libknet/compress.c

```diff
@@ -32,7 +32,7 @@
  * Always add new items before the last NULL.
  */
 
-static compress_model_t compress_modules_cmds[] = {
+static compress_model_t compress_modules_cmds[KNET_MAX_COMPRESS_METHODS + 1] = {
```

Comment:
Are you sure this is really needed? The array will autoinitialize because of last line

Reply:
according to coverity scan the size of the array is the number of entries in the array, so 12 atm. When looping around it complains that it might be possible access out-of-boundaries.
Pre-allocating the array will make looping safer.

Reply:
But the array is preallocated because of last line in the array initializer.

Reply:
This was discussed on IRC. Part of the problem is coverity cleverness (removing idx < KNET_MAX_COMPRESS_METHODS would work same way).